build/webservices-js-templates-generator/src/main/java/com/liferay/ide/servlet/WebServicesServlet.java

```diff
@@ -0,0 +1,205 @@
+import com.liferay.portal.kernel.util.CharPool;
+import com.liferay.portal.kernel.util.FileUtil;
+import com.liferay.portal.kernel.util.MethodParameter;
+import com.liferay.portal.kernel.util.StringPool;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.xml.Document;
+import com.liferay.portal.kernel.xml.Element;
+import com.liferay.portal.kernel.xml.SAXReaderUtil;
+
+/**
+ * @author Terry Jia
+ */
+public class WebServicesServlet extends HttpServlet {
+
+	protected Map<String, Set<JSONWebServiceActionMapping>> getJSONWebServiceClazz() {
+		List<JSONWebServiceActionMapping> jsonWebServiceActionMappings =
+			JSONWebServiceActionsManagerUtil.getJSONWebServiceActionMappings(
+				StringPool.BLANK);
+
+		Map<String, Set<JSONWebServiceActionMapping>> jsonWebServiceClazz =
+			new LinkedHashMap<String, Set<JSONWebServiceActionMapping>>();
+
+		for (JSONWebServiceActionMapping jsonWebServiceActionMapping :
+			jsonWebServiceActionMappings) {
+
+			Class<?> actionClass = jsonWebServiceActionMapping.getActionClass();
+
+			String actionClassName = actionClass.getSimpleName();
+
+			if (actionClassName.endsWith("ServiceUtil")) {
+				actionClassName = actionClassName.substring(0,
+						actionClassName.length() - 11);
+			}
+
+			Set<JSONWebServiceActionMapping> jsonWebServiceMappings =
+				jsonWebServiceClazz.get(actionClassName);
+
+			if (Validator.isNull(jsonWebServiceMappings)) {
+				jsonWebServiceMappings =
+					new LinkedHashSet<JSONWebServiceActionMapping>();
+
+				jsonWebServiceClazz.put(
+					actionClassName, jsonWebServiceMappings);
+			}
+
+			jsonWebServiceMappings.add(jsonWebServiceActionMapping);
+		}
+
+		return jsonWebServiceClazz;
+	}
+
+	protected String getWebServicesXML() {
+		Map<String, Set<JSONWebServiceActionMapping>> jsonWebServiceClazz =
+			getJSONWebServiceClazz();
+
+		Document document = SAXReaderUtil.createDocument("UTF-8");
+
+		Element root = SAXReaderUtil.createElement("templates");
+
+		document.add(root);
+
+		for (String jsonWebServiceClassName : jsonWebServiceClazz.keySet()) {
+			Set<JSONWebServiceActionMapping> jsonWebServiceMappings =
+				jsonWebServiceClazz.get(jsonWebServiceClassName);
+
+			String className = jsonWebServiceClassName;
+
+			if (className.endsWith("Impl")) {
+				className = className.substring(0, className.length() - 4);
+			}
+
+			if (className.endsWith("Service")) {
+				className = className.substring(0, className.length() - 7);
+			}
+
+			for (JSONWebServiceActionMapping jsonWebServiceActionMapping :
+				jsonWebServiceMappings) {
+
+				Element element = SAXReaderUtil.createElement("template");
+
+				String path = jsonWebServiceActionMapping.getPath();
+
+				int pos = path.lastIndexOf(CharPool.SLASH);
+
+				String actionName = path.substring(pos + 1);
+
+				element.add(SAXReaderUtil.createAttribute(
+					element, "name",
+					"liferay-webservices-" + className + "-" + actionName));
```

Comment:
Lets name the templates "jsonws-" the other is too hard to remember and type in.